processor/filterprocessor/testdata/config_expr.yaml

```diff
@@ -0,0 +1,41 @@
+receivers:
+    examplereceiver:
+
+processors:
+    filter/empty:
+        metrics:
+            include:
+                match_type: expr
+    filter/include:
+        metrics:
+            include:
+                match_type: expr
+                expressions:
+                    - Label("foo") == "bar"
+                    - HasLabel("baz")
```

Comment:
Do we need to support multiple expressions? This is equivalent to AND-ing the expressions, right? Should we just allow a single expressions, like this:
```yaml
                match_type: expr
                expression: Label("foo") == "bar" && HasLabel("baz")
```

Reply:
Yes, we could just support a single expression. But the idea was to try to be consistent with the other match types, which do support multiple statements. So, for example, the following two filter processors have the same effect:

```
processors:
  filter/expr:
    metrics:
      exclude:
        match_type: expr
        expressions:
          - MetricName == "system.cpu.time"
          - MetricName == "system.disk.io"
  filter/strict:
    metrics:
      exclude:
        match_type: strict
        metric_names:
          - system.cpu.time
          - system.disk.io
```

I think when users think of the expressions in English they might think of them as AND, as in "I want to exclude metrics named 'system.cpu.time' and I want to exclude metrics named 'system.disk.io'", but since the conditions are evaluated for each metric or each datapoint, they are effectively ORed, as in "Is this metric/datapoint named 'system.cpu.time' or is it named 'system.disk.io'? If so, exclude it."

I think in practice users are going to want to filter out "abusive" metrics and set the expression to something like `MetricName == "some.metric" && Label("mylabel") == "some_value"` then come back later and filter out another abusive metric. I'm thinking that adding another expression requires less thought than figuring out the logical grouping and logical operator, and ultimately ending up with a possibly large, compound statement. Having said that, if we allow multiple expressions, users could still create a single large expression if they wanted.

Either way, I can take a look at the README and see if it could be updated to make this clear.

Reply:
Yes, please add an example in README to make it clear. Otherwise LGTM but would also want @jrcamp to review the PR.